axelor-sale/src/main/java/com/axelor/apps/sale/service/SaleOrderGeneratorServiceImpl.java

```diff
@@ -94,4 +104,29 @@ protected void checkClientPartner(Partner clientPartner, SaleOrder saleOrder)
           company.getName());
     }
   }
+
+  private void setInAti(Boolean inAti, SaleOrder saleOrder) throws AxelorException {
+    if (inAti != null) {
+      checkinAti(saleOrder);
+      saleOrder.setInAti(inAti);
+    }
+  }
+
+  protected void checkinAti(SaleOrder saleOrder) throws AxelorException {
+
+    Company company = saleOrder.getCompany();
+
+    if (saleConfigRepository
+                .find(saleConfigService.getSaleConfig(company).getId())
+                .getSaleOrderInAtiSelect()
+            == 1
+        || saleConfigRepository
+                .find(saleConfigService.getSaleConfig(company).getId())
+                .getSaleOrderInAtiSelect()
+            == 2) {
```

Comment:
Please check SaleOrderInitValueService#getInAti to see how to use SaleConfigService.